purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -672,26 +707,44 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
         identityManager.currentAppUserID.let { appUserID ->
             purchases.forEach { purchase ->
                 if (purchase.containedPurchase.purchaseState == Purchase.PurchaseState.PURCHASED) {
-                    backend.postReceiptData(
-                        purchase.purchaseToken,
-                        appUserID,
-                        purchase.sku,
-                        allowSharingPlayStoreAccount,
-                        purchase.presentedOfferingIdentifier,
-                        !consumeAllTransactions,
-                        { info ->
-                            consumeAndSave(consumeAllTransactions, purchase)
-                            cachePurchaserInfo(info)
-                            sendUpdatedPurchaserInfoToDelegateIfChanged(info)
-                            onSuccess?.let { it(purchase, info) }
-                        }, { error, errorIsFinishable ->
-                            if (errorIsFinishable) {
+                    val postToBackend: (SkuDetails?) -> Unit = { skuDetails ->
+                        backend.postReceiptData(
+                            purchase.purchaseToken,
+                            appUserID,
+                            purchase.sku,
+                            allowSharingPlayStoreAccount,
+                            purchase.presentedOfferingIdentifier,
+                            !consumeAllTransactions,
+                            skuDetails?.priceAmountMicros?.div(1000000.0),
+                            skuDetails?.priceCurrencyCode,
+                            { info ->
                                 consumeAndSave(consumeAllTransactions, purchase)
-                            }
-                            onError?.let { it(purchase, error) }
-                        })
+                                cachePurchaserInfo(info)
+                                sendUpdatedPurchaserInfoToDelegateIfChanged(info)
+                                onSuccess?.let { it(purchase, info) }
+                            }, { error, errorIsFinishable ->
+                                if (errorIsFinishable) {
+                                    consumeAndSave(consumeAllTransactions, purchase)
+                                }
+                                onError?.let { it(purchase, error) }
+                            })
+                    }
+                    if (purchase.type == PurchaseType.INAPP) {
+                        billingWrapper.querySkuDetailsAsync(
+                            BillingClient.SkuType.INAPP,
+                            listOf(purchase.sku),
+                            { skuDetailsList ->
+                                postToBackend(skuDetailsList.first { it.sku == purchase.sku })
+                            },
+                            { postToBackend(null) }
+                        )
+                    } else {
+                        postToBackend(null)
```

Comment:
We only post to backend the price and currency if it's INAPP. I want to default to not posting it if it fails fetching the SkuDetails. I will change the method postToBackend to default to null so that it's more clear.